internal/sdk/cloudian/sdk.go

```diff
@@ -405,6 +419,44 @@ func (client Client) GetGroup(ctx context.Context, groupID string) (*Group, erro
 	}
 }
 
+func percentage(value uint64, percent float64) uint64 { // nolint:unparam
+	return uint64(float64(value) * percent)
+}
+
+func qosQueryMap(user User, qos QoS) map[string]string {
+	return map[string]string{
+		"userId":               user.UserID,
+		"groupId":              user.GroupID,
+		"hlStorageQuotaKBytes": qos.StorageQuota.KBString(),
+		"wlStorageQuotaKBytes": ByteSize(percentage(qos.StorageQuota.KB(), 0.75)).KBString(),
+		"hlStorageQuotaCount":  fmt.Sprintf("%d", qos.StorageQuotaCount),
+		"wlStorageQuotaCount":  fmt.Sprintf("%d", percentage(qos.StorageQuotaCount, 0.75)),
+		"hlRequestRate":        fmt.Sprintf("%d", qos.RequestRate),
+		"wlRequestRate":        fmt.Sprintf("%d", percentage(qos.RequestRate, 0.75)),
+		"hlDataKBytesIn":       qos.DataRateInbound.KBString(),
+		"wlDataKBytesIn":       ByteSize(percentage(qos.DataRateInbound.KB(), 0.75)).KBString(),
+		"hlDataKBytesOut":      qos.DataRateOutbound.KBString(),
+		"wlDataKBytesOut":      ByteSize(percentage(qos.DataRateOutbound.KB(), 0.75)).KBString(),
```

Comment:
Shouldn't the qos struct have all the configurable fields? Can we start simple and just serialize the input struct with basic types onto the post body?

Reply:
As the ints are represented as strings, we would need an internal struct to convert to, which is then serialized. Like we did with Group.

Reply:
Unfortunately not a POST body in this case, but 12 mandatory query params instead!
Pushed an updated suggestion where warning limits are configurable by user